lang/typescript/src/utils/regions.ts

```diff
@@ -46,3 +99,100 @@ export function getRegionConfig(countryCode: string): RegionYamlConfig | null {
 
   return null;
 }
+
+export function addressUsesScript(
+  fieldDefinition: FieldConcatenationRule[],
+  address: Partial<Address>,
+  script: Script,
+): boolean {
+  const charsetFound = fieldDefinition.reduce((matchesCharset, field) => {
+    if (address[field.key]) {
+      const value = address[field.key];
+      if (value) {
+        return matchesCharset || stringUsesScript(value, script);
+      }
+    }
+
+    return matchesCharset;
+  }, false);
+
+  return charsetFound;
+}
```

Comment:
This is the logic to check if any field in the definition matches a particular script ("Han" for Chinese). If we find anything we know we should use the `zh-TW` version.